### ISSUE TITLE
It returns to its original state 30hz when cable disconnected from the monitor

### ISSUE THREAD

**Issue by Zarpar** [Floris497/mac-pixel-clock-patch-V2#134]
1. 2016 new MacBook m7
2. macOS Sierra 10.12.2
3. SIP disabled
4. obviously patched ( Core Display )
5. monitor : LG 27ud88 ( 4k, 60hz, usb-c )

At first, Always thank you for your efforts.
It works with 60hz very well when turning on ( starting up ) my computer and connecting cable, but that' only one time. Disconnecting usb-c cable and connect again, Strangely, it takes a little longer time than prev ( first time connecting ) and changes to 30hz. And then, I must restart my computer for using 60hz again. I've discovered the same issue on this forum but that's not helping me. I've tried all I can do but no return.

**Comment by Floris497** [comment]
I have to say that I have the smallest idea if what's going on.. and I can't really test since I don't have the hardware on hand :/ makes it i easier if I would have.. I will think what can be happening here but I'm not very familiar with the whole USB-C thing, it's a really confusing port because it supports so many different things besides USB. So at this moment I don't know :/

**Comment by Zarpar** [comment]
Thank you!
I will try your assumption with another cables and inform you.

**Comment by mscharnowski** [comment]
Zarpar, same here. Just got my 27UD88 two days ago, using it with a 2016 Macbook. Patch worked fine using the latest files found here, but then, upon working offsite and re-connecting the monitor, it went back to 30hz. 

Hope we can get this sorted, as running that thing on 60hz is a core requirement for me.

Cheers!

**Comment by Floris497** [comment]
I guess rebooting is the 'quick' fix now if I'm not wrong. Again, it's not that easy for me to understand the problem without seeing it :)
I'm also sure the nearest Apple Store wouldn't like me messing with their HW :D

I'll report when something new comes to mind about this issue..

**Comment by mscharnowski** [comment]
Nope, rebooting does not work. Tried it multiple times, tried re-applying the patch (terminal says "system patched"), but nothing I can do will make the monitor go back to 60hz at this time.

**Comment by Floris497** [comment]
Okay that's interesting.. and unpatch reboot patch reboot? 
Rebooting with and without cable inserted? 
I don't understand how it can only work once.. :/

**Comment by mscharnowski** [comment]
How do i unpatch? 

Is the process intended to be done with the monitor plugged in, or plugging it in afterwards?

**Comment by Floris497** [comment]
You can unpatch using the unpatch command please see the readme :) and i guess it doesn't matter if your screen is pluged in or not when patching. So what was the process of getting 60Hz working in the first place?

**Comment by mscharnowski** [comment]
I just followed the instructions from 9to5mac, except I used the new CoreDisplay instead of IOKit. Monitor was plugged in all the time, worked with 60hz until I disconnected and connected again.

**Comment by Floris497** [comment]
Did you reboot after patching? I'm quite sure it's necessary just want to confirm

**Comment by mscharnowski** [comment]
Yes, I did.

**Comment by Floris497** [comment]
Did you already messed around a bit with switchresx?

**Comment by mscharnowski** [comment]
Nope, let me try.

**Comment by Floris497** [comment]
Let me know what comes from that :)

**Comment by mscharnowski** [comment]
Wow - that actually worked. It's not back to 60hz...let's see if this is stable. Thanks!

**Comment by Floris497** [comment]
Oh great :) let me know if you run into further problems!

**Comment by mscharnowski** [comment]
Floris, a possibly dumb question: Why wouldn't I only use SwitchResX from the beginning? What is it that your patch does? Does it enable my macbook to drive the monitor at 60hz, and SwitchResX then activates that?

**Comment by Floris497** [comment]
Some lock is set in software for some reason, might be a performance thing or inherited from older standards. It might be that your mac can do without. But I don't think so. You could try. What this patch does it jumps over the code that checks if your current pixel clock (resolution*refresrate/s) is below this value. To see it a bit simplified. What it does is that after patching it does recognize the newly available refresh rates available. But maybe it looses this late for some reason. Might have to do with your displays utilized protecols. Switchresx will overwrite EDID data an thus make the mac think this screen supports it. If the patch isn't done switchresx will most probably say something like 'invalid' hope that answers your question

**Comment by mscharnowski** [comment]
Yup. Thanks for your great work!

**Comment by mscharnowski** [comment]
Floris, just turned on the screen and laptop in the morning at the same time, and the 60hz option has disappeared from the SwitchResX menu. Just FYI, will keep fiddling around. 

<img width="263" alt="screenshot 2017-01-20 08 35 22" src="https://cloud.githubusercontent.com/assets/25206298/22157250/c9a4283e-deeb-11e6-92d5-480b5258d7a4.png">

**Comment by mscharnowski** [comment]
...and if you then turn off the external monitor and turn it back on, all options are here again.

Strange stuff. 

<img width="263" alt="screenshot 2017-01-20 10 10 32" src="https://cloud.githubusercontent.com/assets/25206298/22160114/d267372e-def8-11e6-9523-889b4a505383.png">

**Comment by Floris497** [comment]
Interesting..

**Comment by Zarpar** [comment]
Hello!
I've tested and just noticed 'reboot' doesn't help me back to 60hz.
As well, changing cable USB-C to HDMI to USB-C make run 60hz without reboot. very weird.
I think I guess there's such a auto function selecting DP alternative mode or not in kernel level... hmm..
**disable 'put hard disk to sleep when possible' solve this phenomenon at least now maybe just for me.**

**Comment by Floris497** [comment]
Nect time it heppens please chech iF HDMI or DP signal usused. This can verry well be the case. Thanks for pointing that out :)
